gstreamer-pbutils/src/auto/discoverer_info.rs

```diff
@@ -0,0 +1,186 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ 47eb915)
+// from gir-files (https://github.com/gtk-rs/gir-files @ ???)
+// DO NOT EDIT
+
+use DiscovererResult;
+use DiscovererSerializeFlags;
+use DiscovererStreamInfo;
+use ffi;
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use gst;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct DiscovererInfo(Object<ffi::GstDiscovererInfo>);
+
+    match fn {
+        get_type => || ffi::gst_discoverer_info_get_type(),
+    }
+}
+
+impl DiscovererInfo {
+    pub fn from_variant(variant: &glib::Variant) -> Option<DiscovererInfo> {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_discoverer_info_from_variant(variant.to_glib_none().0))
+        }
+    }
+}
+
+unsafe impl Send for DiscovererInfo {}
+unsafe impl Sync for DiscovererInfo {}
+
+pub trait DiscovererInfoExt {
+    fn copy(&self) -> DiscovererInfo;
+
+    fn get_audio_streams(&self) -> Vec<DiscovererStreamInfo>;
+
+    fn get_container_streams(&self) -> Vec<DiscovererStreamInfo>;
+
+    fn get_duration(&self) -> gst::ClockTime;
+
+    #[cfg(any(feature = "v1_14", feature = "dox"))]
+    fn get_live(&self) -> bool;
+
+    fn get_misc(&self) -> Option<gst::Structure>;
+
+    fn get_missing_elements_installer_details(&self) -> Vec<String>;
+
+    fn get_result(&self) -> DiscovererResult;
+
+    fn get_seekable(&self) -> bool;
+
+    fn get_stream_info(&self) -> Option<DiscovererStreamInfo>;
+
+    fn get_stream_list(&self) -> Vec<DiscovererStreamInfo>;
+
+    fn get_streams(&self, streamtype: glib::types::Type) -> Vec<DiscovererStreamInfo>;
+
+    fn get_subtitle_streams(&self) -> Vec<DiscovererStreamInfo>;
+
+    fn get_tags(&self) -> Option<gst::TagList>;
+
+    fn get_toc(&self) -> Option<gst::Toc>;
+
+    fn get_uri(&self) -> Option<String>;
```

Comment:
It seems like we don't have guards for NULL URIs on the input and it will lead to an Error. So this get needs to deal with that. The other option is to add checks at the API for rejecting NULLs, if that can be considered a non-breaking change.